backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/utils/ZonedDateTimeCodecTest.java

```diff
@@ -0,0 +1,133 @@
+import static com.datastax.driver.core.querybuilder.QueryBuilder.select;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.ZonedDateTime;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.james.backends.cassandra.CassandraCluster;
+import org.apache.james.backends.cassandra.components.CassandraIndex;
+import org.apache.james.backends.cassandra.components.CassandraModule;
+import org.apache.james.backends.cassandra.components.CassandraTable;
+import org.apache.james.backends.cassandra.components.CassandraType;
+import org.apache.james.backends.cassandra.init.CassandraModuleComposite;
+import org.apache.james.backends.cassandra.init.CassandraZonedDateTimeModule;
+import org.assertj.core.util.Lists;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import com.datastax.driver.core.schemabuilder.SchemaBuilder;
+import com.google.common.collect.ImmutableList;
+
+@RunWith(Parameterized.class)
+public class ZonedDateTimeCodecTest {
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> getParameters() {
+        return Lists.newArrayList(new Object[] {null},
+            new Object[] {ZonedDateTime.parse("2016-04-07T02:01Z")},
+            new Object[] {ZonedDateTime.parse("2016-04-07T02:01:05+07:00")},
+            new Object[] {ZonedDateTime.parse("2016-04-07T02:01:05+07:00[Asia/Vientiane]")});
+    }
+
+    private final ZonedDateTime zonedDateTime;
+
+    private ZonedDateTimeCodec zonedDateTimeCodec;
+    private CassandraCluster cassandra;
+
+    public ZonedDateTimeCodecTest(ZonedDateTime zonedDateTime) {
+        this.zonedDateTime = zonedDateTime;
+    }
+
+    @Before
+    public void setUp() {
+        cassandra = CassandraCluster.create(
+            new CassandraModuleComposite(
+                new CassandraZonedDateTimeModule(),
+                new DateRowModule()));
+
+        zonedDateTimeCodec = new ZonedDateTimeCodec(cassandra.getTypesProvider().getDefinedUserType(CassandraZonedDateTimeModule.ZONED_DATE_TIME));
+        cassandra.getConf()
+            .getCluster()
+            .getConfiguration()
+            .getCodecRegistry()
+            .register(zonedDateTimeCodec);
+    }
+
+    @After
+    public void tearDown() {
+        cassandra.clearAllTables();
+    }
+
+    @Test
+    public void zonedDateTimeShouldBeRetrieved() {
+        cassandra.getConf().execute(insertInto(DateRowModule.TABLE_NAME)
+            .value(DateRowModule.KEY, 18)
+            .value(DateRowModule.DATE, zonedDateTime));
+        ZonedDateTime retrievedZonedDateTime = cassandra.getConf()
+            .execute(select()
+                .from(DateRowModule.TABLE_NAME)
+                .where(eq(DateRowModule.KEY, 18)))
+            .one()
+            .get(DateRowModule.DATE, zonedDateTimeCodec);
+        if (zonedDateTime == null) {
```

Comment:
If the tool is not the good one, change the tool, not the way you are supposed to write a test.